app/controllers/mainstream_browse_pages_controller.rb

```diff
@@ -4,6 +4,35 @@ class MainstreamBrowsePagesController < ApplicationController
 
   before_filter :require_gds_editor_permissions!
 
+  def edit
+    @topics = Topic.all
+  end
+
+  def show; end
+
+  def update
+    # Convert the String ids to Topic objects so that
+    # `update_attributes` correct updates and associates
+    # them with the given `MainstreamBrowsePage` object.
+    if params.require(:mainstream_browse_page).key? :topics
+      topic_ids = params.require(:mainstream_browse_page)[:topics]
+      topics = topic_ids.reject(&:blank?).map { |t| Topic.find(t) }
+      msb_params = tag_params.merge({"topics" => topics})
+    else
+      msb_params = tag_params
+    end
+
+    if @resource.update_attributes(msb_params)
+      PanopticonNotifier.update_tag(
+        presenter_klass.new(@resource)
+      )
+
+      redirect_to mainstream_browse_page_path(@resource)
+    else
+      render :edit
```

Comment:
This will need `@topics` assigned like the edit action, otherwise the view won't work correctly.